### PR TITLE
🚧 Update proxy config

### PR DESCRIPTION
**Pull request by timja**
Note currently this change is demonstrating the test passes for the core PR

Here is our checklist for contributors. No hard requirement here, just a reminder

- [x] Please describe what you did
Added a test for proxy config exporting, updated config based on upstream change in jenkins core https://github.com/jenkinsci/jenkins/pull/3935

- [ ] Link to issue you're working on if there's a relevant one

- [x] Did you provide a test-case to demonstrate feature actually works / issue is fixed ?

- [x] Please also consider adding a line to [CHANGELOG.md](https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by timja** [comment]
not really a WIP 😉 all the code is there.

**Comment by casz** [comment]
Well your PR title says otherwise :|

**Comment by timja** [comment]
It's demonstrating a test for a change in core, but requires a core incrementals version.

**Comment by casz** [comment]
oh, proxy appears under jenkins now when it is databound? Even after removing the proxy configurator, well gosh darned it :smile:

**Comment by timja** [comment]
Yeah, turns out `jenkins` is anything under the `Jenkins` instance

File: plugin/src/test/java/io/jenkins/plugins/casc/core/ProxyConfiguratorTest.java [hunk 1]
```diff
@@ -61,4 +68,41 @@ public void shouldSetProxyWithMinimumFields() throws Exception {
...
+
+    @Test
+    @ConfiguredWithCode("Proxy.yml")
+    public void describeProxyConfig() throws Exception {
+        final JenkinsConfigurator root = getJenkinsConfigurator();
+        ConfiguratorRegistry registry = ConfiguratorRegistry.get();
+        ConfigurationContext context = new ConfigurationContext(registry);
+        final CNode configNode = getProxyNode(root, context);
+
+        Secret password = Secret.decrypt(getProxyNode(root, context).asMapping().getScalarValue("password"));
```
[the remark this answers is not in the document]
Comment (timja) [reply plugin/src/test/java/io/jenkins/plugins/casc/core/ProxyConfiguratorTest.java]: Think I tried that and it seemed to remove it from the node, can double check later on

**Comment by timja** [comment]
Currently failing on:
```
io.jenkins.plugins.casc.BackwardCompatibilityTest.should_accept_legacy_symbols_on_descriptors(io.jenkins.plugins.casc.BackwardCompatibilityTest)

[2019-03-23T11:37:08.714Z] [ERROR]   Run 1: BackwardCompatibilityTest.should_accept_legacy_symbols_on_descriptors:35 expected:<...solete, please use '[dumb]'> but was:<...solete, please use '[permanent]'>

[2019-03-23T11:37:08.714Z] [ERROR]   Run 2: BackwardCompatibilityTest.should_accept_legacy_symbols_on_descriptors:35 expected:<...solete, please use '[dumb]'> but was:<...solete, please use '[permanent]'>

[2019-03-23T11:37:08.714Z] [ERROR]   Run 3: BackwardCompatibilityTest.should_accept_legacy_symbols_on_descriptors:35 expected:<...solete, please use '[dumb]'> but was:<...solete, please use '[permanent]'>

[2019-03-23T11:37:08.714Z] [ERROR]   Run 4: BackwardCompatibilityTest.should_accept_legacy_symbols_on_descriptors:35 expected:<...solete, please use '[dumb]'> but was:<...solete, please use '[permanent]'>

[2019-03-23T11:37:08.714Z] [ERROR]   Run 5: BackwardCompatibilityTest.should_accept_legacy_symbols_on_descriptors:35 expected:<...solete, please use '[dumb]'> but was:<...solete, please use '[permanent]'>

```
the proxy tests

and a spotbugs error in `AdminWhitelistRuleConfigurator`

**Comment by casz** [comment]
You should rebase on my work in the branch Jenkinsv161

**Comment by casz** [comment]
I will update it to v164.1

**Comment by timja** [comment]
Will rebase after your merge, the changes in this branch make the update work
Proxy test is failing for some reason

Debugging it

**Comment by casz** [comment]
I have no plans to merge #790 just yet :cold_sweat: we have yet to have a proper discussion about bumping core.

I suggest however you target the JenkinsV164.1 branch and rebase your work on that one.

**Comment by timja** [comment]
I have no plans on merging this branch ever, its just for demonstrating a core change works.

Some parts of this will be extracted if / when the core change is merged

**Comment by casz** [comment]
:grimacing:

**Comment by oleg-nenashev** [comment]
Bots vs. Maintainers :P

**Comment by casz** [comment]
👏

**Pull request closed without merging**